### ISSUE TITLE
Etapa 4

### ISSUE THREAD

**Issue by JairDantas** [JairDantas/Data_Science_GTI#7]

**Comment by JairDantas** [comment]
import pandas as pd
import matplotlib.pyplot as plt
df = pd.read_csv('/meu_dataset2.csv')
print(df.head())
print(df.info())



[meu_dataset2.csv](https://github.com/JairDantas/Data_Science_GTI/files/14939539/meu_dataset2.csv)

**Comment by JairDantas** [comment]
AULA 17.04

import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.cluster import KMeans
from sklearn.metrics import accuracy_score
# Carregar os dados do CSV para um DataFrame do Pandas
dados = pd.read_csv('/content/meu_dataset2.csv')
# Análise Exploratória de Dados (EDA)
# Verificar estatísticas descritivas dos dados
print(dados.describe())
# Visualizar a distribuição das respostas de cada coluna
for coluna in dados.columns:
 plt.figure(figsize=(8, 6))
 sns.histplot(dados[coluna], bins=10, kde=True)
 plt.title(f'Distribuição de {coluna}')
 plt.xlabel(coluna)
 plt.ylabel('Frequência')
 plt.show()
# Dividir os dados em variáveis independentes (X) e variável dependente (y)
X = dados[['A Faculdade oferece um bom local de estudos', 'Recomendaria a faculdade']]
y = dados['Curso']
# Dividir os dados em conjunto de treino e teste
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)
# Regressão Logística
# Criar e treinar o modelo de regressão logística
modelo_regressao_logistica = LogisticRegression()
modelo_regressao_logistica.fit(X_train, y_train)
# Fazer previsões no conjunto de teste
y_pred_regressao_logistica = modelo_regressao_logistica.predict(X_test)
# Avaliar o desempenho do modelo de regressão logística
acuracia_regressao_logistica = accuracy_score(y_test, y_pred_regressao_logistica)
print(f'Acurácia da Regressão Logística: {acuracia_regressao_logistica:.2f}')
# K-Means
# Selecionar apenas as variáveis numéricas para o algoritmo K-Means
X_kmeans = dados[['Ano de Graduação', 'A Faculdade oferece um bom local de estudos', 'Curso']]
# Escolher o número de clusters (k)
k = 3
# Criar e treinar o modelo K-Means
modelo_kmeans = KMeans(n_clusters=k, random_state=42)
modelo_kmeans.fit(X_kmeans)
# Atribuir cada ponto de dados a um cluster
labels_kmeans = modelo_kmeans.labels_
# Visualizar os clusters
plt.figure(figsize=(8, 6))
sns.scatterplot(data=dados, x='A Faculdade oferece um bom local de estudos', y='Curso', hue=labels_kmeans, palette='Set1', legend='full')
plt.title('Clusters de Satisfação e Frequência de Uso')
plt.xlabel('Satisfação')
plt.ylabel('Frequência de Uso')
plt.show()

**Comment by JairDantas** [comment]
AULA 24.04.24

segue em anexo codigo python 



[Documento sem título.docx](https://github.com/JairDantas/Data_Science_GTI/files/15102247/Documento.sem.titulo.docx)

____________________________________________________________________________________________________________________________

codigo: 

01 - # Importando a biblioteca Pandas que faz o tratamento de dados
# Importar dados a partir do arquivo CSV e transformar em seu DataSet
# Criar a variável dataset_unifecaf que recebe os dados importados
import pandas as pd
caminho_arquivo = "/content/DataSet-ML-24042024_NEW.csv"
dados = pd.read_csv(caminho_arquivo)
# Verificar tipo de dados das colunas
print("\nTipos de dados das colunas:")
print(dados.dtypes)
# Verificar valores ausentes
print("\nValores ausentes:")
print(dados.isnull().sum())
# Remover linhas com valores ausentes (se necessário)
dados.dropna(inplace=True)
# Verificar as primeiras linhas do conjunto de dados
print("\nPrimerias 5 linhas do conjunto de dados:")
print(dados.head())
# Verificar informações sobre o conjunto de dados
print("\nInformações sobre o conjunto de dados:")
print(dados.info())
# Verificar estatísticas descritivas do conjunto de dados
print("\nEstatísticas descritivas do conjunto de dados:")
print(dados.describe())
# Verificar valores únicos em cada coluna categórica
print("\nValores únicos em cada coluna categórica:")
for coluna in dados.select_dtypes(include='object').columns:
      print(f"\n{coluna}:")
      print(dados[coluna].unique())

_________________________________________________________________________
02 - # Converta colunas não numéricas em numéricas, se possível, ou descarte-as
for col in dados.columns:
   try:
       dados[col] = pd.to_numeric(dados[col])
   except ValueError:
       dados = dados.drop(col, axis=1)


# Agora você pode calcular a matriz de correlação
matriz_correlacao = dados.corr()


# E visualize isso
plt.figure(figsize=(10, 8))
sns.heatmap(matriz_correlacao, annot=True, cmap='coolwarm', fmt=".2f", linewidths=.5)
plt.title('Matriz de Correlação')
plt.show()

sns.heatmap(matriz_correlacao, annot=True, cmap='coolwarm', fmt=".2f", linewidths=.5)
plt.title('Matriz de Correlação')
plt.show()

_________________________________________________________________________
03 -  # Gráfico de barras da contagem de 'Motivo_Evasao'
plt.figure(figsize=(8, 6))
sns.countplot(x='Motivo_Evasao', data=dados,palette='muted')
plt.title('Contagem de Motivo_Evasao')
plt.xlabel('Motivo_Evasao')
plt.ylabel('Contagem')
plt.xticks(rotation=45)
plt.show()
# Gráfico de dispersão entre 'Frequencia' e 'Pontuacao_Satisfacao'
plt.figure(figsize=(8, 6))
plt.scatter(dataset_unifecaf['Frequencia'],
dataset_unifecaf['Pontuacao_Satisfacao'], color='blue', alpha=0.5)
plt.title('Gráfico de Dispersão: Frequencia vs Pontuacao_Satisfacao')
plt.xlabel('Frequencia')
plt.ylabel('Pontuacao_Satisfacao')
plt.show()
_________________________________________________________________________
04  - from sklearn.cluster import KMeans
import matplotlib.pyplot as plt
# Definir o número de clusters
n_clusters = 3 # Defina o número de clusters desejado
# Selecionar apenas as colunas relevantes para o algoritmo K-Means
X = dados[['Nota1', 'Nota2']]
# Criar o modelo K-Means
modelo_kmeans = KMeans(n_clusters=n_clusters, random_state=42)
# Treinar o modelo
modelo_kmeans.fit(X)
# Obter os rótulos dos clusters para cada amostra
rotulos_clusters = modelo_kmeans.labels_
# Obter as coordenadas dos centróides de cada cluster
centroides = modelo_kmeans.cluster_centers_
# Plotar os clusters
plt.figure(figsize=(10, 6))
# Plotar os pontos de dados de cada cluster
for cluster in range(n_clusters):
plt.scatter(X[rotulos_clusters == cluster]['Nota1'], X[rotulos_clusters == cluster]['Nota2'], label=f'Cluster {cluster+1}')
# Plotar os centróides de cada cluster
plt.scatter(centroides[:, 0], centroides[:, 1], color='black',
marker='x', label='Centróides')
# Adicionar rótulos e legendas
plt.title('Clusters de Notas')
plt.xlabel('Nota1')
plt.ylabel('Nota2')
plt.legend()
plt.grid(True)
# Mostrar o gráfico
plt.show()
